utils/common/src/main/java/org/apache/brooklyn/util/javalang/coerce/TypeCoercerExtensible.java

```diff
@@ -119,8 +120,23 @@ public static TypeCoercerExtensible newEmpty() {
         if (targetTypeToken.getType() instanceof ParameterizedType) {
             if (value instanceof Collection && Collection.class.isAssignableFrom(targetType)) {
                 result = tryCoerceCollection(value, targetTypeToken, targetType);
+                
+                if (result != null && result.isAbsent() && targetType.isInstance(value)) {
+                    log.warn("Failed to coerce collection from " + value.getClass().getName() + " to " + targetTypeToken  
+                            + "; returning uncoerced result to preserve (deprecated) backwards compatibility", 
+                            Maybe.getException(result));
+                    return Maybe.of((T)result);
```

Comment:
Should be `return Maybe.of( (T) value );`, but I don't understand how it's different from letting the code execution continue. Same for change below.